src/s390x/dasd.rs

```diff
@@ -0,0 +1,284 @@
+        .chain_err(|| format!("setting block size {}", bytes))?;
+    ctx.assign_device(&image, true)
+        .chain_err(|| format!("opening image {}", image.display()))?;
+
+    let mut dd_options = Vec::new();
+    let mut partitions = Vec::new();
+    let mut start = 2; //the first 2 tracks of the ECKD DASD are reserved;
+    let mut table = ctx.get_partitions().chain_err(|| "reading partitions")?;
+    let mut count = table.nents();
+
+    for mut pt in &mut table {
+        let uuid = pt.uuid().unwrap_or_else(|_| "UUID unknown".to_string());
+        let skip = pt.start().chain_err(|| format!("{}: no start", uuid))?;
+        let size = pt.size().chain_err(|| format!("{}: no size", uuid))?;
+        let seek = blocks * start;
+        let end = (size + blocks) / 12;
+        eprintln!("{} start: {} size: {}", uuid, skip, size);
+        let mut opts = Opts::default();
+        opts.input_block_size = bytes as usize;
+        opts.output_block_size = bytes as usize;
+        opts.iflags = dd_lib::opts::IFlag::FULLBLOCK;
+        opts.input_file = Some(image.to_str().unwrap().to_string());
+        opts.output_file = Some(dasd.to_string());
+        opts.input_seek = Some(skip as usize);
+        opts.count = Some(size as usize);
+        opts.output_seek = Some(seek as usize);
+        dd_options.push(opts);
+
+        count -= 1;
+        if count != 0 {
+            partitions.push(format!("[{}, {}, native]", start, end));
+        } else {
+            partitions.push(format!("[{}, last, native]", start));
+        };
+        start = end + 1;
+    }
+    Ok((dd_options, partitions))
+}
+
+/// Get disk bus id
+///
+/// # Arguments
+/// * `dasd` - dasd device, i.e. smth like /dev/dasda
+fn bus_id(dasd: &str) -> Result<String> {
+    let cmd = Command::new("lszdev")
+        .arg("-n")
+        .arg("-c")
+        .arg("ID")
+        .arg("--by-node")
+        .arg(dasd)
+        .output()
+        .chain_err(|| "failed to execute lszdev")?;
+    let s = cmd
+        .stdout
+        .iter()
+        .map(|&c| c as char)
+        .filter(|&c| c != '\n' && c != '\r')
+        .collect::<String>();
+    eprintln!("Bus ID: {}", s);
+    Ok(s)
+}
+
+/// Check if disk is already formatted or not
+///
+/// # Arguments
+/// * `dasd` - dasd device, i.e. smth like /dev/dasda
+fn is_unformatted(dasd: &str) -> Result<bool> {
+    let id = bus_id(dasd)?;
+    let path = format!("/sys/bus/ccw/devices/{}/status", id);
+    let contents = read_to_string(&path).chain_err(|| format!("reading {}", path))?;
+    Ok(contents.contains("unformatted"))
+}
+
+/// Check if disk is valid or not
+///
+/// # Arguments
+/// * `dasd` - dasd device, i.e. smth like /dev/dasda
+fn is_invalid(dasd: &str) -> Result<bool> {
+    let cmd = Command::new("fdasd").arg("-p").arg(dasd).output()?;
+    let output = cmd.stdout.iter().map(|&c| c as char).collect::<String>();
+    Ok(output.contains("disk label block is invalid"))
```

Comment:
Does `fdasd` output localized strings?

Reply:
it's hardcoded c-string:
 `printf("\nCannot show requested information because "
			       "the disk label block is invalid\n");`

Reply:
In cases where we care about specific output strings, we should run the program with the `LC_ALL` environment variable set to `C` in case localization is added in the future.